### ISSUE TITLE
OneofValidator should allow oneof fields to be empty

### ISSUE THREAD

**Issue by yuri-sergiichuk** [SpineEventEngine/base#378]
The current implementation of the `OneofValidator` does not allow **any** `oneof` field to be not set.

The expected behavior is that it is possible to have unset `oneof` fields, e.g. we have an `origin_id` fields in the `EventContext` message that may be empty by its contract:
```protobuf
// Meta-information for an event.
message EventContext {
    ...
    oneof origin {
        // The context of the command which generated this event.
        CommandContext command_context = 2 [(required) = true];

        // The context of the event which generated this event.
        EventContext event_context = 6 [(required) = true];

        // The event was imported using this `ActorContext`.
        ActorContext import_context = 11 [(required) = true];
    }
    ...
    // An ID of the origin messsage.
    //
    // The case of this `oneof` is always the same as the case of `origin` (except for
    // the `import_context`, which does not have a corresponding field in `origin_id`).
    //
    // The `origin` and `origin_id` groups are not merged for the reasons of binary compatibility.
    //
    oneof origin_id {

        // The ID of the origin command.
        CommandId command_id = 12;

        // The ID of the origin event.
        EventId event_id = 13;
    }
    ...
}
```

**Comment by yuri-sergiichuk** [comment]
As discussed vocally, the only way to make `oneof` field required should be to use `(required_field` option as follows:
```protobuf
message EventContext {
    option (required_field) = "command_context | event_context | import_context";
    ...
    oneof origin {
        // The context of the command which generated this event.
        CommandContext command_context = 2;

        // The context of the event which generated this event.
        EventContext event_context = 6;

        // The event was imported using this `ActorContext`.
        ActorContext import_context = 11;
    }
    ...
}

The `required` option at the message field level for `oneof` fields should be treated now as a configuration error.

Also, by default, `oneof` fields should **not** be required

**Comment by alexander-yevsyukov** [comment]
@yuri-sergiichuk please see if #521 addresses this issue. If so, please close.

**Comment by yuri-sergiichuk** [comment]
Superseded by #521 and fixed in #523 (the new syntax described in the PR).
